service/src/main/kotlin/fi/espoo/evaka/calendarevent/CalendarEventQueries.kt

```diff
@@ -67,11 +61,7 @@ JOIN calendar_event ce ON cea.calendar_event_id = ce.id
 LEFT JOIN daycare_group dg ON dg.id = cea.group_id
 LEFT JOIN person p ON p.id = cea.child_id
 LEFT JOIN calendar_event_time cet ON cet.calendar_event_id = ce.id
-WHERE (:calendarEventId IS NULL OR ce.id = :calendarEventId) 
-AND (:unitId IS NULL OR cea.unit_id = :unitId) 
-AND (:groupId IS NULL OR cea.group_id = :groupId)
-AND (:range IS NULL OR ce.period && :range)
-AND (:eventTypes::calendar_event_type[] IS NULL OR ce.event_type = ANY(:eventTypes::calendar_event_type[]))
+WHERE ${predicate(where.forTable(""))}
```

Comment:
I couldn't really find or understand the implementation for this `forTable`. I wanted to double check how does it work when passing an empty string. Where should I look?

Reply:
When you say `Predicate { sql("...") }`, the passed table name is available as `it`, or `$it` in string interpolations. `it` is actually just the default name for the sole parameter of the unnamed function.

The `Predicate { ... }` syntax calls `Predicate` with a function as the sole parameter. In this case, the call to `Predicate` is "delegated" to the companion object's invoke function here:

https://github.com/espoon-voltti/evaka/blob/95e6dfbdadc413166172366d54570b86666605ff/service/src/main/kotlin/fi/espoo/evaka/shared/db/Predicate.kt#L70-L71

As you can see, it just constructs a `Single` object.

Here's the implementation of `forTable` for `Single`. It calls the wrapped function passing the table name to it:
https://github.com/espoon-voltti/evaka/blob/95e6dfbdadc413166172366d54570b86666605ff/service/src/main/kotlin/fi/espoo/evaka/shared/db/Predicate.kt#L46-L47

In this PR, an empty string is passed, because multiple tables are involved so there's no single table name to pass and `$it` is not used.